dcmgr/lib/dcmgr/models/network.rb

```diff
@@ -154,45 +154,137 @@ def ipv4_u32_dynamic_range_array
     end
 
     def add_ipv4_dynamic_range(range_begin, range_end)
-      range_begin = IPAddress::IPv4.new("#{range_begin}/#{self[:prefix]}")
-      range_end = IPAddress::IPv4.new("#{range_end}/#{self[:prefix]}")
-      test_inclusion(*validate_range_args(range_begin, range_end)) { |range, op|
-         case op
-         when :coverbegin
-           range.range_end = range_begin
-         when :coverend
-           range.range_begin = range_end
-         when :inccur
-           range.destroy
-         end
-         range.save_changes
-      }
+      range_begin_u32, range_end_u32 = validate_range_args(range_begin, range_end)
+      dhcp_range_dataset.included_ranges(range_begin_u32, range_end_u32).destroy
```

Comment:
1. Will be rolled back if the error is a Ruby Exception. 
2. when scheduler iterates the rows, it misses to take exclusive locks on rows. so it is possible to cause typical transaction isolation problem. we need to change scheduler side too as they are readers who expect set of rows are consistent.
    https://github.com/axsh/wakame-vdc/blob/master/dcmgr/lib/dcmgr/scheduler/ip_address/incremental.rb#L79